Workflow - retry option

CahidArda opened the pull request:
Adds retry option to serve parameters:
```
export const POST = serve(
  ...,
  {
    retries: 2
  }
)
```

fahreddinozcan commented:
I guess the default retry count is `3`. We can maybe keep this value in a single constant like `DEFAULT_RETRY_COUNT`, so that if we ever change it, it's easier to change? also more understandable that it's the default count when reading

fahreddinozcan commented:
Is it possible to create a test that asserts the function is not retried the 3rd time, if `retry: 2`

CahidArda commented:
added constant `DEFAULT_RETRIES` and added a test with retry: 2.

The test is an integration test which doesn't run on github CI sadly (because it needs to run against qstash server). It can be run locally though and passes.

Apart from this test, I also check the Upstash-Retries header in multiple tests. I think it's sufficient for the test of the sdk. It's upto the qstash server at that point

@fahreddinozcan
